### PR TITLE
Fix mistake observability maturity conflict override

### PR DESCRIPTION
**Pull request by rhetoric101**
When I was working on previous changes in #4614, I had a git conflict and mistakenly overrode changes in #4625.

This PR corrects those capitalization overwrites, as well as changes the name of the file to "introduction.mdx" instead of "index.mds." The name change is because that file is no longer on a parent node.

The reviewer should make sure that the following two headings are now sentence case in the navigation:

"Observability maturity"
" Uptime, performance & reliability"

Also, please confirm that the redirect from the old index.mdx resolves to the introduction topic: /docs/new-relic-solutions/observability-maturity

### CODE REVIEW

File: src/nav/new-relic-solutions.yml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 title: Guides and best practices
 path: /docs/new-relic-solutions
 pages:
-  - title: Observability Maturity
+  - title: Observability maturity
     pages:
       - title: Introduction
         path: /docs/new-relic-solutions/observability-maturity
-      - title: Uptime, Performance & Reliability
+      - title: Uptime, performance & reliability
```
Comment (barbnewrelic) [inline src/nav/new-relic-solutions.yml:8]: I wonder if we need to change the ampersand to "and"?

Comment (barbnewrelic) [reply src/nav/new-relic-solutions.yml:8]: In Gatsby build, the "Observability maturity" dropdown works just fine!. But When I click "Introduction" it goes to the former landing page just fine but that doc title still shows "Observability Maturity" capital M. Intentional? https://docswebsitedevelop-rhsfixobsmaturityconflict.gtsb.io/docs/new-relic-solutions/observability-maturity/introduction/  
I also checked all the dropdowns and links from this page. T and they go to the right docs. Thanks for taking care of this!

**Review by barbnewrelic — APPROVED** [review]
@rhetoric101  I added a few reviewer questions/comments, take 'em or leave 'em! Then feel free to use your admin privileges to merge this PR. No additional review required. Thanks for taking care of this taxonomy fix!

File: src/nav/new-relic-solutions.yml (hunk 1 above)
Comment (rhetoric101) [reply src/nav/new-relic-solutions.yml:8]: @barbnewrelic, I changed that heading to "Observability maturity" to follow our style guide.

**Review by barbnewrelic — APPROVED** [review]
Looks great!

**Pull request merged**